manim/scene/scene.py

```diff
@@ -863,6 +864,18 @@ def add_static_frames(self, duration):
             num_frames=int(duration / dt),
         )
 
+    def add_click_sound(self, **kwargs):
+        """Add a click sound to the animation.
+
+        .. seealso::
+
+            :meth:`~.add_sound`
+        """
+        path_sound = (
+            Path(__file__).parent.parent.parent / "manim" / "assets" / "click.wav"
```

Comment:
Navigating around with `.parent.parent.parent` still seems *wrong* and very fragile to me.

I'd like to get the opinion of someone experienced with packaging (maybe @naveen521kk ?): It's not clear to me that adding a non-python file to the python package is as simple as adding it somewhere inside the `manim` directory. Will that not cause problems for the next release?

Reply:
> 
> 
> Navigating around with `.parent.parent.parent` still seems _wrong_ and very fragile to me.

This seems to be a common way to do this as [per this post](https://stackoverflow.com/a/53465812/10530136). Even the top answer does something similar without pathlib by using a file at the top level directory for the project (so `constants.py` would make the most sense).

> I'd like to get the opinion of someone experienced with packaging (maybe @naveen521kk ?): It's not clear to me that adding a non-python file to the python package is as simple as adding it somewhere inside the `manim` directory. Will that not cause problems for the next release?

I'm not sure about this. But if this is an issue, we could always remove these and just provide the link to the resource file as we already are in the docs.

Reply:
> I'd like to get the opinion of someone experienced with packaging (maybe @naveen521kk ?): It's not clear to me that adding a non-python file to the python package is as simple as adding it somewhere inside the manim directory. Will that not cause problems for the next release?

Docs here: https://python-poetry.org/docs/pyproject/#include-and-exclude

Usually, we don't need to do anything because poetry works out of the box by looking over `.gitignore` files. (That's a reason we are using poetry to poetry :wink: )

Also, I don't think we would need to include sound files, rather maybe a Plugin can subclass this and have those files? Thoughts?

> Navigating around with .parent.parent.parent still seems wrong and very fragile to me.

You should use `.resolve()` to go above a parent directory.

Reply:
> You should use `.resolve()` to go above a parent directory.

If we are going to include sound files in the PyPI manim package, I think we should do the following:
1. Add `ROOT_DIR = Path(__file__).resolve().parent` to `manim/constants.py`
2. Use [git-lfs](https://git-lfs.github.com/).

But if we're going to use a plugin (which would be more valuable to host galleries of mp3/svg/png images) then this file can be removed as well as the test_case.

Reply:
> I'm in favor of a plugin solution/keeping the core smaller btw.
In case that you mean "smaller" according to "less diskspace", I just reduced the size of the audio file by a factor of 100 by changing the format from wav to mp3. (now it is about 2 kB)

> Also, I don't think we would need to include sound files, rather maybe a Plugin can subclass this and have those files? Thoughts?

Yes, we don't need many sound files, and this pr is also not about providing a huge sound library for videos.
It is about providing one sound file for debugging scenes when exact timing is important. I think this sound file can be very powerful when it comes to testing if some elements in your scene are in or out of sync to each other.

Another idea: As this sound file is really really small, we could as well just store it as an numpy array.

Reply:
Seems fine to have a single file, I have seen matplotlib and similar having many font files, and it seems it wouldn't hurt use to do so.

> Another idea: As this sound file is really really small, we could as well just store it as an numpy array.

Are we allowed to modify and distribute the file?

Reply:
> Are we allowed to modify and distribute the file?

Yes, it's Creative Commons.

> > I'm in favor of a plugin solution/keeping the core smaller btw.
> 
> In case you mean "smaller" according to "less disk space", I just reduced the size of the audio file by a factor of 100 by changing the format from wav to mp3. (now it is about 2 kB)

I do mean that, and that's great! Totally forgot this was an uncompressed .wav file.

> Another idea: As this sound file is really really small, we could as well just store it as an numpy array.

I don't think that's necessary.

> Yes, we don't need many sound files, and this pr is also not about providing a huge sound library for videos.
> It is about providing one sound file for debugging scenes when exact timing is important. I think this sound file can be very powerful when it comes to testing if some elements in your scene are in or out of sync to each other.

I'm just weary of having a gallery in the core when this seems apt for a plugin. A couple is fine for I guess "essential sounds" to demonstrate the library's core works (let's just keep them as compressed MP3s). Let's consider this the sound equivalent of a VMobject (like Circle(), Square(), etc...) and only have a few.